### ISSUE TITLE
Metric for CRAN check results

### ISSUE THREAD

**Issue by mpadge** [pharmaR/riskmetric#125]
Motivated by the now fairly stable interface provided by rOpenSci's [`cchecks` package](https://github.com/ropenscilabs/cchecks) which archives all results of CRAN checks at [`cranchecks.info`](https://docs.cranchecks.info/). This would admittedly be restricted to CRAN packages only, and so would exclude bioconductor, but these checks provide very useful diagnostic indicators. The following compares [`foghorn`](https://github.com/fmichonneau/foghorn) (current check results only) output with `cchecks` (by default last 30 days of checks):

``` r
library(foghorn)
pkg <- "rstan"
res1 <- cran_details(pkg = "rstan")
chk <- apply (res1, 1, function (i) {
              print (cli::boxx (paste0 (i [["result"]],
                                        " (N = ", i [["n_flavors"]], ")"),
                                padding = c (0, 5, 0, 5)))
              message (i [["message"]])
})
#> ┌───────────────────────┐
#> │     NOTE (N = 12)     │
#> └───────────────────────┘
#>      GNU make is a SystemRequirements.
#> ┌───────────────────────┐
#> │     NOTE (N =  7)     │
#> └───────────────────────┘
#>      installed size is 192.0Mb
#>      sub-directories of 1Mb or more:
#>      libs  189.2Mb

library(cchecks)
x <- cch_pkgs_history (x = pkg)
h <- x$data$history$check_details # main data.frame of check results
table (data.frame (output = h$output,
                   n_flavors = vapply (h$flavors, length, integer (1))))
#>                                    n_flavors
#> output                              12
#>   GNU make is a SystemRequirements. 30
```

<sup>Created on 2020-06-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

The 30 values for `ccheck` results are the last 30 days of checks, with that Note occurring 12 times on each day. ("installed size" Notes do not appear in `ccheck` results for some reason?). One of the primary motivations of `cchecks` is because CRAN often update their checks without warning, with authors only getting an email sometime after packages start failing. Accessing these services could provide an additional useful diagnostic.

**Comment by dgkf** [comment]
Sounds great. Part of the roadmap is certainly to work the typical checks into the mix. This is an additional permutation of that problem giving some history of the check results. 

Do you see this as an alternative to the typical `R CMD check` or an addition? My initial impression is that this might be a suitable replacement for running checks locally for a `cran_remote` `pkg_ref`.

**Comment by mpadge** [comment]
I see it as an addition, because:

1. `R CMD check` will generally only be run on one local machine by users of `riskmetric`, whereas the full CRAN checks are run on all of their machines; and
2. A local check will only ever check using locally installed versions of all software, which may be older than current CRAN versions.

There's also the additional advantage of CRAN UBSAN and ASAN checks which are not run as part of a standard `R CMD check`, so CRAN checks can pick up more problems than local ones generally do.

In short, it's a view into the future, to indicate any potential problems which may arise, yet not be detected via a local `R CMD check`. That said, you could of course quite easily use this as a substitute for time-consuming local checks for `cran_remote` packages, as you suggest.

**Comment by emilliman5** [comment]
#170  and #169 have implemented both running R CMD check locally and scraping the R CMD check results from CRAN and Bioc.
